xbmc/platform/darwin/tvos/tvosShared.m

```diff
@@ -18,36 +18,38 @@ + (NSString*)getSharedID
 + (NSURL*)getSharedURL
 {
   NSString* sharedID = [self getSharedID];
-  if ([self isJailbroken])
-    return [[NSURL fileURLWithPath:@"/var/mobile/Library/Caches"]
-        URLByAppendingPathComponent:sharedID];
-  else
+  if ([self isSandboxed])
   {
     NSFileManager* fileManager = [NSFileManager defaultManager];
     NSURL* sharedUrl = [fileManager containerURLForSecurityApplicationGroupIdentifier:sharedID];
+    // e.g. /private/var/mobile/Containers/Shared/AppGroup/32B9DA1F-3B1F-4DBC-8326-ABB08BF16EC9/
     sharedUrl = [sharedUrl URLByAppendingPathComponent:@"Library" isDirectory:YES];
     sharedUrl = [sharedUrl URLByAppendingPathComponent:@"Caches" isDirectory:YES];
+    // e.g.
+    // /private/var/mobile/Containers/Shared/AppGroup/32B9DA1F-3B1F-4DBC-8326-ABB08BF16EC9/Library/Caches/
     return sharedUrl;
   }
+  else
+  {
+    return [[NSURL fileURLWithPath:@"/var/mobile/Library/Caches"]
+        URLByAppendingPathComponent:sharedID];
+  }
 }
 
-+ (BOOL)IsTVOSSandboxed
++ (BOOL)isSandboxed
 {
   // @todo merge with CDarwinUtils::IsIosSandboxed
   static BOOL ret;
   static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
-    // we re NOT sandboxed if we are installed in /var/mobile/Applications with greeng0blin jailbreak
+    // we re NOT sandboxed if we are installed in /var/mobile/Applications with greeng0blin
+    // jailbreak
+    // TODO: Check is this check still works on recent jailbreaks (checkra1n, ChimeraTV, unc0ver)
     ret = ![[self mainAppBundle].bundlePath containsString:@"/var/mobile/Applications/"];
   });
   return ret;
 }
```

Comment:
Do we want to look at generalising this into the the ios sandbox check? Or are they different enough, its just easier keeping them separate?

https://github.com/xbmc/xbmc/blob/5d9c54b88747d321dcbbba29f124c7cc7f3b6867/xbmc/platform/darwin/DarwinUtils.mm#L171

Reply:
Yes you are right, I will try to use the sandbox check from `DarwinUtils.mm`.

Reply:
For this one, I'am not sure if what I am doing is good.
If we want to use `CDarwinUtils::IsIosSandboxed()` in [tvosShared.m](https://github.com/xbmc/xbmc/blob/master/xbmc/platform/darwin/tvos/tvosShared.m) we need to modify [ExtraTargets.cmake](https://github.com/xbmc/xbmc/blob/master/cmake/scripts/darwin_embedded/ExtraTargets.cmake) and add in SOURCES and HEADERS DarwinUtils, Application, DllPaths, ... without those changes I get Linker error when Xcode build `kodi-topshelf` target.
Maybe I'am wrong?

Reply:
@SylvainCecchetto you could just move that method to another file to compile less stuff for top shelf, if that makes sense. But yes, you must include `CDarwinUtils` to SOURCES and HEADERS.

Reply:
Maybe a file in `ios-common` folder should be better for hosting `IsIosSandboxed()` function?